api/client/src/main/java/org/projectnessie/client/http/ResponseContext.java

```diff
@@ -16,17 +16,17 @@
 package org.projectnessie.client.http;
 
 import java.io.IOException;
-import java.io.InputStream;
 import java.net.URI;
+import org.projectnessie.client.rest.io.CapturingInputStream;
 
 /** Interface for the important parts of a response. This is created after executing the request. */
 public interface ResponseContext {
 
   Status getResponseCode() throws IOException;
 
-  InputStream getInputStream() throws IOException;
+  CapturingInputStream getInputStream() throws IOException;
```

Comment:
TBH, this change will affect _all_ requests, not just the ones for OAuth, so I'm really not a fan of this change. `OAuth2ClientConfig` already does a `new CapturingInputStream()` - so it would be a `CapturingInputStream` around a `CapturingInputStream`?

Reply:
Sorry I forgot to change `OAuth2ClientConfig` after changing `ResponseContext`.

That said, you have a point that the cost in space of this change is probably too high.

Reply:
I reverted that part of the change and reduced the scope of this change to just the oauth client.